storage/remote/queue_manager.go

```diff
@@ -1661,33 +1662,43 @@ func populateTimeSeries(batch []timeSeries, pendingData []prompb.TimeSeries, sen
 
 func (s *shards) sendSamples(ctx context.Context, samples []prompb.TimeSeries, sampleCount, exemplarCount, histogramCount int, pBuf *proto.Buffer, buf *[]byte, enc Compression) error {
 	begin := time.Now()
-	err := s.sendSamplesWithBackoff(ctx, samples, sampleCount, exemplarCount, histogramCount, 0, pBuf, buf, enc)
-	s.updateMetrics(ctx, err, sampleCount, exemplarCount, histogramCount, 0, time.Since(begin))
+	rs, err := s.sendSamplesWithBackoff(ctx, samples, sampleCount, exemplarCount, histogramCount, 0, pBuf, buf, enc)
+	s.updateMetrics(ctx, err, sampleCount, exemplarCount, histogramCount, 0, rs, time.Since(begin))
```

Comment:
I like that we would be capturing and using the stats if they exist for a v1 request. We should be sure we document this change clearly in the changelog, just to avoid any confusion if someone's still using v1 and then all of a sudden their failed datapoints metrics start to look very different because their receiver has started supporting the response stat headers for v1.

Reply:
Added to the changelog!